node/src/isolate/external.cpp

```diff
@@ -229,11 +233,25 @@ struct external_t::inner_t :
     }
 
     void reconnect(const std::error_code& e) {
-        if(session) {
-            COCAINE_LOG_INFO(log, "reconnecting to external isolation daemon");
-            session->detach(e);
-            session.reset();
-            connect();
+        if(!connecting) {
+            connecting = true;
+            if(session) {
+                session->detach(e);
+                session.reset();
+            }
+
+            auto reconnect_timeout_ms = args.as_object().at("reconnect_timeout_ms", 3000u).as_uint();
+            COCAINE_LOG_INFO(log, "queueing reconnect to external isolation daemon after {} ms", reconnect_timeout_ms);
+
+            connect_timer.expires_from_now(boost::posix_time::milliseconds(reconnect_timeout_ms));
+            std::weak_ptr<inner_t> weak_self(shared_from_this());
+            connect_timer.async_wait([=](const std::error_code&) {
```

Comment:
"This can fail for other reasons - man timerfd."
http://think-async.com/Asio/asio-1.10.6/doc/asio/reference/basic_deadline_timer/async_wait.html
 For each call to async_wait(), the supplied handler will be called exactly once. The handler will be called when:

    The timer has expired.
    The timer was cancelled, in which case the handler is passed the error code asio::error::operation_aborted.
Anyway we don't care in ANY case. We just will try to connect a bit earlier this time.
And there is no call of deleted method. Async wait is called from living object and it looks like handler won't be called if timer is destroyed